superset/assets/javascripts/explore/AdhocMetric.js

```diff
@@ -0,0 +1,37 @@
+export default class AdhocMetric {
+  constructor(adhocMetric) {
+    this.column = adhocMetric.column;
+    this.aggregate = adhocMetric.aggregate;
+    this.hasCustomLabel = !!(adhocMetric.hasCustomLabel && adhocMetric.label);
+    this.fromFormData = !!adhocMetric.optionName;
+
+    if (this.hasCustomLabel) {
+      this.label = adhocMetric.label;
+    } else {
+      this.label = this.getDefaultLabel();
+    }
+
+    this.optionName = adhocMetric.optionName ||
+      `metric_${Math.random().toString(36).substring(2, 15)}_${Math.random().toString(36).substring(2, 15)}`;
```

Comment:
is there a util func to generate ids that we could use?

Reply:
this is what my internet research suggested i do 🤕